### PR TITLE
Fix overriding of alpine builder

### PR DESCRIPTION
**Pull request by natalieparellano**
We need --pull-policy=always when the builder contains extensions (the builder must exist in a registry so that we can pull the manifest) 

However pulling the remote builder overrides the locally built one, making our test useless and causing us to re-publish the builder that was already in Docker Hub 

Spinning up a local registry should solve the problem

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -39,20 +37,26 @@ build-builder-bionic: build-linux-packages build-sample-root
 build-linux-buildpacks: build-buildpacks-alpine build-buildpacks-bionic
 
 build-buildpacks-alpine: build-sample-root
+	@echo "> Starting local registry to store alpine builder (when builder contains extensions it must exist in a registry so that builds can use --pull-policy=always and we don't want to override the locally built builder)"
+	docker run -d --rm -p 5000:5000 registry:2
+	sleep 2
+	docker tag cnbs/sample-builder:alpine localhost:5000/cnbs/sample-builder:alpine
+	docker push localhost:5000/cnbs/sample-builder:alpine
+
```
Comment (AidanDelaney) [inline Makefile:45]: Is this a good use-case for `docker/podman-compose`?  We could spin up `registry:2` in one container, and the application builds in another?  This would avoid the need for `--network=host`.

**Review by AidanDelaney — COMMENTED** [review]
These changes work, so I'm happy to land them.  I have two questions to improve my understanding.

File: Makefile [hunk 2]
```diff
@@ -29,8 +29,6 @@ build-linux-builders: build-builder-alpine build-builder-bionic
 build-builder-alpine: build-linux-packages build-sample-root
 	@echo "> Building 'alpine' builder..."
 	$(PACK_CMD) builder create cnbs/sample-builder:alpine --config $(SAMPLES_ROOT)/builders/alpine/builder.toml $(PULL_POLICY_NEVER)
-	docker run cnbs/sample-builder:alpine ls /cnb/extensions/samples_curl
```
[the remark this answers is not in the document]
Comment (natalieparellano) [reply Makefile:32]: Ah well, I guess we could keep the checks if we want - they were added previously because I was pulling the sample builder and wondering, what could possibly be going on?

Thinking about this more, the alpine builder in its current state is actually broken 😞 for [demo purposes](https://github.com/buildpacks/docs/blob/main/content/docs/extension-author-guide/create-extension/why-dockerfiles.md#build-the-application-image) which is not ideal. I think I will simplify the tutorial (and make the builder working by default) by introducing a `BP_` environment variable to control the buildpack behavior WRT extensions.

TLDR; I'll update this PR and create a corresponding docs PR to make everything better

File: Makefile [hunk 3]
```diff
@@ -39,20 +37,26 @@ build-builder-bionic: build-linux-packages build-sample-root
...
-	$(PACK_CMD) build sample-hello-processes-app:alpine -v --builder cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/hello-processes $(PACK_BUILD_FLAGS)
+	$(PACK_CMD) build sample-hello-processes-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/hello-processes --network=host
 
 	@echo "> Creating 'hello-world' app using 'alpine' builder..."
-	$(PACK_CMD) build sample-hello-world-app:alpine -v --builder cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/hello-world $(PACK_BUILD_FLAGS)
+	$(PACK_CMD) build sample-hello-world-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/hello-world --network=host
 
 	@echo "> Creating 'java-maven' app using 'alpine' builder..."
-	$(PACK_CMD) build sample-java-maven-app:alpine -v --builder cnbs/sample-builder:alpine --path apps/java-maven $(PACK_BUILD_FLAGS)
+	$(PACK_CMD) build sample-java-maven-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/java-maven --path apps/java-maven --network=host
```
Comment (natalieparellano) [inline Makefile:56]:
```suggestion
	$(PACK_CMD) build sample-java-maven-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --path apps/java-maven --network=host
```

File: Makefile [hunk 4]
```diff
@@ -39,20 +37,26 @@ build-builder-bionic: build-linux-packages build-sample-root
...
 
 	@echo "> Creating 'kotlin-gradle' app using 'alpine' builder..."
-	$(PACK_CMD) build sample-kotlin-gradle-app:alpine -v --builder cnbs/sample-builder:alpine --path apps/kotlin-gradle $(PACK_BUILD_FLAGS)
+	$(PACK_CMD) build sample-kotlin-gradle-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --buildpack $(SAMPLES_ROOT)/buildpacks/kotlin-gradle --path apps/kotlin-gradle --network=host
```
Comment (natalieparellano) [inline Makefile:59]:
```suggestion
	$(PACK_CMD) build sample-kotlin-gradle-app:alpine -v --builder localhost:5000/cnbs/sample-builder:alpine --path apps/kotlin-gradle --network=host
```

File: Makefile (hunk 2 above)
[the remark this answers is not in the document]
Comment (natalieparellano) [reply Makefile:32]: Added the aforementioned improvements in cc5328a and https://github.com/buildpacks/docs/pull/571

File: Makefile (hunk 1 above)
Comment (natalieparellano) [reply Makefile:45]: Maybe... do you think the benefit is worth the additional overhead?

Comment (AidanDelaney) [reply Makefile:45]: No. Let's go with it the way it is.

**Comment by natalieparellano** [comment]
CI failing with `failed to fetch builder image 'index.docker.io/cnbs/sample-builder:jammy'` - we can re-kick it when that deploy succeeds

File: buildpacks/hello-extensions/bin/build [hunk 5]
```diff
@@ -7,7 +7,7 @@ tree --version
 
 layers_dir=$1
 
-cat >> "${layers_dir}/launch.toml" <<EOL
+cat >>"${layers_dir}/launch.toml" <<EOL
```
Comment (natalieparellano) [inline buildpacks/hello-extensions/bin/build:10]:
```suggestion
cat >> "${layers_dir}/launch.toml" <<EOL
```

**Review by AidanDelaney — APPROVED** [review]
LGTM.
